### PR TITLE
feat: add key method

### PR DESCRIPTION
**Pull request by polothy**
*Issue #, if available:* https://github.com/aws-cloudformation/cloudformation-guard/issues/476

*Description of changes:*

100% done by GitHub Copilot, I just kept clicking buttons till it said it was done 😄 I don't know Rust at all, so I cannot even tell if this is garbage or not, but I did see the tests run via Copilot.

---
*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: guard/src/rules/eval_context.rs [hunk 1]
```diff
@@ -1194,6 +1194,7 @@ pub(crate) enum FunctionName {
     ToLower,
     ToUpper,
     UrlDecode,
+    Key, // <-- add this line
```
Comment (polothy) [inline guard/src/rules/eval_context.rs:1197]:
```suggestion
    Key,
```

File: guard/src/rules/eval_context.rs [hunk 2]
```diff
@@ -1213,6 +1214,7 @@ impl FunctionName {
             | FunctionName::ParseEpoch
             | FunctionName::ParseChar => 1,
             FunctionName::Now => 0,
+            FunctionName::Key => 1, // key expects 1 argument
```
Comment (polothy) [inline guard/src/rules/eval_context.rs:1217]:
```suggestion
            FunctionName::Key => 1,
```

**Review by satyakigh — COMMENTED** [review]
`key()` silently drops every input after the first. `key.rs` returns from inside the first loop iteration, and `KeyFunction::call` wraps that single value, so `key(%buckets)` for multiple resources yields only the first logical ID even though the linked issue asks for the key of node(s). That can hide later non-compliant resources. Please make this an element-wise projection (one result per `QueryResult`, with empty input producing no results) and add a multi-resource end-to-end test; the current tests cover only one value and the fabricated empty-string fallback.

**Comment by polothy** [comment]
Let AI push a new re-based commit. The change looks more complete, but as stated before, I don't know Rust (still haven't learned it in a year hah). It did say it was able to re-produce your problem, so I'm assuming the tests/etc address it.

Is there a status on the project? Seems like it went dormant for quite some time. This bug, I have been trying to get it fixed since 2022 (see https://github.com/aws-cloudformation/cloudformation-guard/issues/267) - even had a meeting with Guard team manager via our AWS TAM. The [aws-guard-rules-registry](https://github.com/aws-cloudformation/aws-guard-rules-registry) seems abandoned, which mostly leaves rule writing to whoever wants to use Guard.
